core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -145,27 +150,47 @@ public void write(MarshalledEntry entry) {
    }
 
    @Override
+   public void writeBatch(Iterable entries) {
```

Comment:
Seems this should be `Iterable<MarshalledEntry>`. Then you don't need the cast below.